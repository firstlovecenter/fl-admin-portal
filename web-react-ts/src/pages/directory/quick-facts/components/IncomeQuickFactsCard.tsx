import React from 'react'
import { Badge } from 'react-bootstrap'
import '../QuickFacts.css'
import { getPercentageChange } from './quick-fact-utils'

export interface IncomeQuickFactsProps {
  incomeDetails: {
    churchType: string
    cardType: string
    leadersName: string
    churchName: string
    churchAvgIncomeThisMonth: number | string
    avgHigherLevelIncomeThisMonth: number | string
    higherLevelName: string
  }[]
}

const IncomeQuickFactsCard = (props: IncomeQuickFactsProps) => {
  const details = props?.incomeDetails[0]

  const percentageRiseOrFall = getPercentageChange(
    details?.churchAvgIncomeThisMonth as number,
    details?.avgHigherLevelIncomeThisMonth as number
  )

  const getBadgeBackground = () => {
    if (percentageRiseOrFall >= 0) return 'green'
    return 'red'
  }

  const getBadgeColor = () => {
    if (percentageRiseOrFall >= 0) return 'badge-percentage-green'
    return 'badge-percentage-red'
  }

  return (
    <div className="w-100 text-center quick-fact-card">
      <div className="church-text">{details?.churchType}</div>
      <div className="stat-text ">
        {' '}
        Average Weekday <br />
        {details?.cardType}{' '}
      </div>
      <div className="leader-text">{details?.leadersName}</div>
      <div className="branch-text">
        {details?.churchName + ' ' + details?.churchType}
      </div>
      <div className="income-number">
        <span className="currency">GHS </span>
        {details?.churchAvgIncomeThisMonth === 'null'
          ? '--'
          : details?.churchAvgIncomeThisMonth}
      </div>
      <div>
        <Badge
          bg={`${getBadgeBackground()}`}
          className={`${getBadgeColor()} mt-auto`}
        >
          {percentageRiseOrFall >= 0 ? '+' : ''}
          {percentageRiseOrFall}%
        </Badge>
      </div>
      <hr className="separator" />
      <div className="income-number">
        <span className="currency">GHS </span>
        {details?.avgHigherLevelIncomeThisMonth === 'null'
          ? '--'
          : details?.avgHigherLevelIncomeThisMonth}
      </div>
      <div className="average-text">
        Average {details?.churchType} <br /> Income
      </div>
      <div className="higher-church-text">{details?.higherLevelName}</div>
    </div>
  )
}

export default IncomeQuickFactsCard
