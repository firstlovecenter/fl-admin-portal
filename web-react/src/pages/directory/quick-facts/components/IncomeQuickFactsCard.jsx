import React from 'react'
import { Badge } from 'react-bootstrap'
import '../QuickFacts.css'

const IncomeQuickFactsCard = (props) => {
  const details = props?.details[0]

  const getPercentageChange = (avgIncome, avgHigherLevelIncome) => {
    var diff = avgIncome - avgHigherLevelIncome
    if (isNaN(diff)) return '--'
    return Math.round((diff / avgIncome) * 100)
  }

  const percentageRiseOrFall = getPercentageChange(
    details?.churchAvgIncomeThisMonth,
    details?.avgHigherLevelIncomeThisMonth
  )

  const getBadgeBackground = () => {
    if (percentageRiseOrFall >= 0) return 'green'
    return 'red'
  }

  const getBadgeColor = () => {
    if (percentageRiseOrFall >= 0) return 'badge-percentage-green'
    return 'badge-percentage-red'
  }

  return (
    <div className="w-100 text-center quick-fact-card">
      <div className="church-text">{details?.churchType}</div>
      <div className="stat-text ">
        {' '}
        Average Weekday <br />
        {details?.cardType}{' '}
      </div>
      <div className="leader-text">{details?.leadersName}</div>
      <div className="branch-text">
        {details?.churchName + ' ' + details?.churchType}
      </div>
      <div className="income-number">
        <span className="currency">GHS </span>
        {details?.churchAvgIncomeThisMonth === 'null'
          ? '--'
          : details?.churchAvgIncomeThisMonth}
      </div>
      <div>
        <Badge
          bg={`${getBadgeBackground()}`}
          className={`${getBadgeColor()} mt-auto`}
        >
          {percentageRiseOrFall >= 0 ? '+' : ''}
          {percentageRiseOrFall}%
        </Badge>
      </div>
      <hr className="separator" />
      <div className="income-number">
        <span className="currency">GHS </span>
        {details?.avgHigherLevelIncomeThisMonth === 'null'
          ? '--'
          : details?.avgHigherLevelIncomeThisMonth}
      </div>
      <div className="average-text">
        Average {details?.churchType} <br /> Income
      </div>
      <div className="higher-church-text">{details?.higherLevelName}</div>
    </div>
  )
}

export default IncomeQuickFactsCard
