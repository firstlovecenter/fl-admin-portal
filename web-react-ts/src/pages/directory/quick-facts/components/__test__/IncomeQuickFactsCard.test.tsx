import { render, cleanup } from '@testing-library/react'
import React from 'react'
import IncomeQuickFactsCard from '../IncomeQuickFactsCard'
import '@testing-library/jest-dom/extend-expect'
import renderer from 'react-test-renderer'

const incomeDetails = [
  {
    churchType: 'Bacenta',
    cardType: 'Income',
    leadersName: 'Edem Ahadzi',
    churchName: 'Madina Bacenta',
    churchAvgIncomeThisMonth: '2330',
    avgHigherLevelIncomeThisMonth: '6380',
    higherLevelName: 'Madina Council',
  },
]

afterEach(cleanup)

it('renders without crashing', () => {
  render(<IncomeQuickFactsCard incomeDetails={incomeDetails} />)
})

it('renders Income Card with correct details', () => {
  const { getByTestId } = render(
    <IncomeQuickFactsCard incomeDetails={incomeDetails} />
  )
  expect(getByTestId('incomeCard')).toHaveTextContent('Income')
})

it('matches snapshot', () => {
  const tree = renderer
    .create(<IncomeQuickFactsCard incomeDetails={incomeDetails} />)
    .toJSON()
  expect(tree).toMatchSnapshot()
})
