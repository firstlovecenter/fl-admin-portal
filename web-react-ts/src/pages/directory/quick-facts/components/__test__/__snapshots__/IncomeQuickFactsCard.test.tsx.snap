// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="w-100 text-center quick-fact-card"
  data-testid="incomeCard"
>
  <div
    className="church-text"
  >
    Bacenta
  </div>
  <div
    className="stat-text "
  >
     
    Average Weekday 
    <br />
    Income
     
  </div>
  <div
    className="leader-text"
  >
    Edem Ahadzi
  </div>
  <div
    className="branch-text"
  >
    Madina Bacenta Bacenta
  </div>
  <div
    className="income-number"
  >
    <span
      className="currency"
    >
      GHS 
    </span>
    2330
  </div>
  <div>
    <span
      className="badge-percentage-red mt-auto badge bg-red"
    >
      -63
      %
    </span>
  </div>
  <hr
    className="separator"
  />
  <div
    className="income-number"
  >
    <span
      className="currency"
    >
      GHS 
    </span>
    6380
  </div>
  <div
    className="average-text"
  >
    Average 
    Bacenta
     
    <br />
     Income
  </div>
  <div
    className="higher-church-text"
  >
    Madina Council
  </div>
</div>
`;
